### PR TITLE
feat: API for deletion of all user favourites

### PR DESCRIPTION
**Pull request by martintomas**
I am adding new API endpoint `DELETE /api/v1/account/users/favourites` which is available to all approved users and which deletes all user's favourites.

Feel free to let me know if you feel that path should look different, I was not 100% sure if putting this under `DELETE` action was good choice.

## Testing instructions

Via Rswag

## Tracking

https://vizzuality.atlassian.net/browse/LET-858

### CODE REVIEW

**Comment by agnessa** [comment]
we need to double check if the intention is to remove all favourites of all sorts, or "remove all projects" separate to "remove all investors"

**Comment by martintomas** [comment]
> we need to double check if the intention is to remove all favourites of all sorts, or "remove all projects" separate to "remove all investors"

Okay, both options are feasible so lets wait for feedback from Su

**Comment by agnessa** [comment]
Susana confirmed this is about removing all sorts of favourites, so let's merge

**Review by agnessa — APPROVED** [review]

File: backend/app/controllers/api/v1/accounts/users_controller.rb [hunk 1]
```diff
@@ -29,6 +29,15 @@ def transfer_ownership
           render json: UserSerializer.new(user, params: {current_user: user}).serializable_hash
         end
 
+        # DELETE
+        def favourites
+          current_user.favourite_projects.destroy_all
```
Comment (tsubik) [inline backend/app/controllers/api/v1/accounts/users_controller.rb:34]: Not better in transaction? Just in `ActiveRecord::Base.transaction`

Comment (martintomas) [reply backend/app/controllers/api/v1/accounts/users_controller.rb:34]: Hmm, yes, it is, I think nothing can blow up here, but just to be on save side ;)

**Pull request merged**